sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStore.scala

```diff
@@ -1173,6 +1203,114 @@ object StateStore extends Logging {
     }
   }
 
+  private def processQueuedProviders(storeConf: StateStoreConf): Unit = {
+    // Get up to 5 queued providers at a time to avoid blocking maintenance for too long
+    val maxProvidersToProcess = 5
+    val providersToProcess = new mutable.ArrayBuffer[StateStoreProviderId](maxProvidersToProcess)
+
+    for (_ <- 1 to maxProvidersToProcess if !partitionsForMaintenance.isEmpty) {
+      val id = partitionsForMaintenance.poll()
+      if (id != null) {
```

Comment:
When can this be `null` ? we are checking for queue being non-empty in the condition above right ?